### ISSUE TITLE
ServiceControl 5.2 and above does not start when configured with AmazonSQS transport

### ISSUE THREAD

**Issue by jpalac** [Particular/ServiceControl#4304]
#### Description

When using ServiceControl 5.2+ with AmazonSQS, the service attempts to start, then throws the following error in the logs and the process dies.

```
2024-06-28 15:14:45.6570|21|Error|ServiceControl.Operations.ErrorIngestion|Error during initial startup attempt for failed message ingestion.|Amazon.SQS.Model.QueueDoesNotExistException: The specified queue does not exist.
 ---> Amazon.Runtime.Internal.HttpErrorResponseException: Exception of type 'Amazon.Runtime.Internal.HttpErrorResponseException' was thrown.
   at Amazon.Runtime.HttpWebRequestMessage.ProcessHttpResponseMessage(HttpResponseMessage responseMessage)
   at Amazon.Runtime.HttpWebRequestMessage.GetResponseAsync(CancellationToken cancellationToken)
   at Amazon.Runtime.Internal.HttpHandler`1.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.Unmarshaller.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.SQS.Internal.ValidationResponseHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.ErrorHandler.InvokeAsync[T](IExecutionContext executionContext)
   --- End of inner exception stack trace ---
   at Amazon.Runtime.Internal.HttpErrorResponseExceptionHandler.HandleExceptionStream(IRequestContext requestContext, IWebResponseData httpErrorResponse, HttpErrorResponseException exception, Stream responseStream)
   at Amazon.Runtime.Internal.HttpErrorResponseExceptionHandler.HandleExceptionAsync(IExecutionContext executionContext, HttpErrorResponseException exception)
   at Amazon.Runtime.Internal.ExceptionHandler`1.HandleAsync(IExecutionContext executionContext, Exception exception)
   at Amazon.Runtime.Internal.ErrorHandler.ProcessExceptionAsync(IExecutionContext executionContext, Exception exception)
   at Amazon.Runtime.Internal.ErrorHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CallbackHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.Signer.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.EndpointDiscoveryHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.EndpointDiscoveryHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CredentialsRetriever.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.RetryHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.RetryHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CallbackHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CallbackHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.ErrorCallbackHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.MetricsHandler.InvokeAsync[T](IExecutionContext executionContext)
   at NServiceBus.Transport.SQS.QueueCache.<>c.<<GetQueueUrl>b__4_0>d.MoveNext() in /_/src/NServiceBus.Transport.SQS/QueueCache.cs:line 38
--- End of stack trace from previous location ---
   at NServiceBus.Transport.SQS.QueueCache.GetQueueUrl(String queueName, CancellationToken cancellationToken) in /_/src/NServiceBus.Transport.SQS/QueueCache.cs:line 34
   at NServiceBus.Transport.SQS.DelayedMessagesPump.Initialize(CancellationToken cancellationToken) in /_/src/NServiceBus.Transport.SQS/DelayedMessagesPump.cs:line 30
   at NServiceBus.Transport.SQS.MessagePump.Initialize(PushRuntimeSettings limitations, OnMessage onMessage, OnError onError, CancellationToken cancellationToken) in /_/src/NServiceBus.Transport.SQS/MessagePump.cs:line 35
   at ServiceControl.Transports.TransportCustomization`1.CreateTransportInfrastructure(String name, TransportSettings transportSettings, OnMessage onMessage, OnError onError, Func`3 onCriticalError, TransportTransactionMode preferredTransactionMode) in /Users/particular/Projects/ServiceControl/src/ServiceControl.Transports/TransportCustomization.cs:line 168
   at ServiceControl.Operations.ErrorIngestion.EnsureStarted(CancellationToken cancellationToken) in /Users/particular/Projects/ServiceControl/src/ServiceControl/Operations/ErrorIngestion.cs:line 145
   at ServiceControl.Operations.ErrorIngestion.EnsureStarted(CancellationToken cancellationToken) in /Users/particular/Projects/ServiceControl/src/ServiceControl/Operations/ErrorIngestion.cs:line 173
   at ServiceControl.Infrastructure.Watchdog.<>c__DisplayClass11_0.<<Start>b__0>d.MoveNext() in /Users/particular/Projects/ServiceControl/src/ServiceControl.Infrastructure/Watchdog.cs:line 52
```

ServiceControl Management utility may still show the service as running.

#### Versions
ServiceControl 5.2+

### Steps to reproduce

Install a new ServiceControl 5.2+ instance or upgrade an existing instance to 5.2+

### Additional Information

This is happening because the AmazonSQS transport is expecting `-delay.fifo` queues for every queue, however these are not created for the queues used by ServiceControl as they are not needed.

#### Workarounds

Manually create the -delay.fifo queues for all the ServiceControl standard queues. This is what that may look like if using Error, Audit and Monitoring instances:

![image](https://github.com/Particular/ServiceControl/assets/4316196/52d7b509-754b-4280-b8e9-8ac1f2f36a1e)

> [!NOTE]
> For any of the above standard queues that you do not see in your setup, you do not need to create the -delay.fifo queue

**Comment by jpalac** [comment]
Work on this issue has started.

**Comment by jpalac** [comment]
Will be shipped with release 5.4.1
